admin-dashboard/src/contexts/AuthContext.tsx

```
@@ -0,0 +1,184 @@
+            setUser(userData);
+            setIsAdmin(userData.role === 'admin');
+          } catch (error) {
+            console.error('Error parsing user data from storage event:', error);
+          }
+        }
+      }
+    };
+
+    window.addEventListener('storage', handleStorageChange);
+    return () => window.removeEventListener('storage', handleStorageChange);
+  }, [accessToken, router, handleLogout]);
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)